torch/csrc/distributed/rpc/rref_impl.h

```diff
@@ -395,6 +399,28 @@ class TORCH_API OwnerRRef final : public RRef {
   friend class RRefContext;
 
   std::shared_ptr<JitFuture> future_;
+
+ public:
+  // Records an event per each stream in the context and stores them in
+  // the current OwnerRRef instance if CUDA is available,
+  // otherwise does nothing.
+  void recordAllStreams(std::shared_ptr<LazyStreamContext> ctx);
+
+  // Blocks all streams in the context on all events previously stored in
+  // the current OwnerRRef instance if CUDA is available,
+  // otherwise does nothing.
+  void blockAllStreams(std::shared_ptr<LazyStreamContext> ctx);
+ private:
+
+ #ifdef USE_CUDA_NOT_ROCM
+  // a storage for CUDA events for synchronization.
+  std::vector<at::cuda::CUDAEvent> cudaEvents_;
+#else
+  // if CUDA is not available we still need a dummy field
+  // to prevent segfaults because of different class size
```

Comment:
Curious, why the different class sizes could be a problem? I thought these are compile time flags, which means when CUDA is not available, it's same as this class does not have this field, no?

Reply:
The practice shows that this header is included to cpp files with CUDA and without and it produces segfaults